optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/listener/support/SupplyVariableListener.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.variable.listener.support;
+
+import java.util.Objects;
+
+import org.optaplanner.core.api.domain.variable.VariableListener;
+import org.optaplanner.core.api.score.director.ScoreDirector;
+import org.optaplanner.core.impl.domain.variable.supply.Supply;
+
+public class SupplyVariableListener<Solution_, Entity_> implements Supply, VariableListener<Solution_, Entity_> {
```

Comment:
Naming wise, it seems more sense to call it `VariableListenerSupply`?

Reply:
Thinking aloud: Supply is an empty interface. Is it worth being an interface?

Reply:
Btw I guess this is an Adaptor or Bridge pattern (GoF), so we should probably name it accordingly.